File: app/src/main/java/com/yannis/thesis/movierecommendationapp/data/UserDao.java
package com.yannis.thesis.movierecommendationapp.data;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import com.yannis.thesis.movierecommendationapp.models.User;
import java.util.List;

@Dao
public interface UserDao {
    @Query("SELECT * FROM User")
    List<User> getAll();
    @Query("SELECT * FROM User WHERE email = :email LIMIT 1")
    User findByEmail(String email);
    @Query("SELECT * FROM User WHERE password = :password LIMIT 1")
    User findByPassword(String password);
    @Query("SELECT * FROM User WHERE id != :id")
    List<User> findAllExcept(String id);
    @Insert
    void insert(User user);
}

File: app/src/main/java/com/yannis/thesis/movierecommendationapp/data/local/UserDao.kt
package com.yannis.thesis.movierecommendationapp.data.local

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.yannis.thesis.movierecommendationapp.data.local.User

@Dao
interface UserDao {
    @Query("SELECT * FROM User")
    fun getAll(): List<User>

    @Query("SELECT * FROM User WHERE email = :email LIMIT 1")
    fun findByEmail(email: String): User?

    @Query("SELECT * FROM User WHERE password = :password LIMIT 1")
    fun findByPassword(password: String): User?

    @Query("SELECT * FROM User WHERE id != :id")
    fun findAllExcept(id: String): List<User>

    @Insert
    fun insert(user: User)
}
